src/configureWorkspace/configureDotNetCore.ts

```diff
@@ -267,20 +257,101 @@ function genDockerFile(serviceNameAndRelativePath: string, platform: Platform, o
+    }
+
+    if (projectFiles.length > 1) {
+        let items = projectFiles.map(p => <vscode.QuickPickItem>{ label: p });
+        let result = await ext.ui.showQuickPick(items, opt);
+        return result.label;
+    } else {
+        return projectFiles[0];
+    }
+}
+
+async function initializeForDebugging(context: IActionContext, folder: WorkspaceFolder, platformOS: PlatformOS, workspaceRelativeDockerfileName: string, workspaceRelativeProjectFileName: string): Promise<void> {
     const scaffoldContext: DockerDebugScaffoldContext = {
         folder: folder,
         platform: 'netCore',
         actionContext: context,
-        dockerfile: dockerfile,
+        // always use posix for debug config because it's committed to source control and works on all OS's
+        // tslint:disable-next-line: no-invalid-template-strings
+        dockerfile: path.posix.join('${workspaceFolder}', workspaceRelativeDockerfileName),
     }
 
     const options: NetCoreScaffoldingOptions = {
-        // always use posix for debug config because it's committed to source control and works on all OS's
         // tslint:disable-next-line: no-invalid-template-strings
-        appProject: path.posix.join('${workspaceFolder}', artifactName),
+        appProject: path.posix.join('${workspaceFolder}', workspaceRelativeProjectFileName),
         platformOS: platformOS,
     }
 
     await dockerDebugScaffoldingProvider.initializeNetCoreForDebugging(scaffoldContext, options);
 }
+
+// tslint:disable-next-line: export-name
+export async function scaffoldNetCore(context: ScaffolderContext): Promise<ScaffoldFile[]> {
+    const os = context.os ?? await context.promptForOS();
+
+    const telemetryProperties = <ConfigureTelemetryProperties>context.telemetry.properties;
+
+    telemetryProperties.configureOs = os;
+
+    const ports = context.ports ?? (context.platform === 'ASP.NET Core' ? await context.promptForPorts([80, 443]) : undefined);
+
+    const rootRelativeProjectFileName = await context.captureStep('project', findCSProjOrFSProjFile)(context.rootFolder);
+    const rootRelativeProjectDirectory = path.dirname(rootRelativeProjectFileName);
+
+    telemetryProperties.packageFileType = path.extname(rootRelativeProjectFileName);
+    telemetryProperties.packageFileSubfolderDepth = getSubfolderDepth(context.rootFolder, rootRelativeProjectFileName);
+
+    const projectFilePath = path.posix.join(context.rootFolder, rootRelativeProjectFileName);
+    const workspaceRelativeProjectFileName = path.posix.relative(context.folder.uri.fsPath, projectFilePath);
+
+    let serviceNameAndPathRelative = rootRelativeProjectFileName.slice(0, -(path.extname(rootRelativeProjectFileName).length));
+    const projFileContents = (await fse.readFile(path.join(context.rootFolder, rootRelativeProjectFileName))).toString();
+
+    // Extract TargetFramework for version
+    const [version] = extractRegExGroups(projFileContents, /<TargetFramework>(.+)<\/TargetFramework/, ['']);
```

Comment:
What if the project is using `TargetFrameworks` property (multiple frameworks)?

Reply:
That's a good point; we simply don't support that yet.  It should be something we can do, but it's worthy of its own issue and PR.  I've added #1532 to track that.